compiler/rustc_middle/src/ty/sty.rs

```diff
@@ -1170,14 +1170,27 @@ impl<'tcx> Ty<'tcx> {
         }
     }
 
-    /// Panics if called on any type other than `Box<T>`.
-    pub fn boxed_ty(self) -> Ty<'tcx> {
+    pub fn boxed_ty(self) -> Option<Ty<'tcx>> {
         match self.kind() {
-            Adt(def, args) if def.is_box() => args.type_at(0),
-            _ => bug!("`boxed_ty` is called on non-box type {:?}", self),
+            Adt(def, args) if def.is_box() => Some(args.type_at(0)),
+            _ => None,
         }
     }
 
+    /// Panics if called on any type other than `Box<T>`.
+    pub fn expect_boxed_ty(self) -> Ty<'tcx> {
+        self.boxed_ty()
+            .unwrap_or_else(|| bug!("`expect_boxed_ty` is called on non-box type {:?}", self))
+    }
+
+    pub fn is_boxed_slice(self) -> bool {
+        self.boxed_ty().is_some_and(Ty::is_slice)
+    }
+
+    pub fn is_boxed_trait(self) -> bool {
+        self.boxed_ty().is_some_and(Ty::is_trait)
+    }
```

Comment:
Please remove these two helpers. They're not generally useful, so code that needs them should just called `boxed_ty().is_some_and(...)`.

Reply:
I've removed `is_boxed_trait` because now that I've unbroken clippy, it's only ever used once.

But `is_boxed_slice` is used 4 times, which is even more than the 2 remaining usages of `expect_boxed_ty`. Should I still remove it?

Reply:
Yeah, I would still prefer if we removed it.